src/libraries/System.Text.Json/tests/Serialization/PropertyNameTests.cs

```diff
@@ -430,6 +430,26 @@ public static void LongPropertyNames(int propertyLength, char ch)
             string jsonRoundTripped = JsonSerializer.Serialize(obj, options);
             Assert.Equal(json, jsonRoundTripped);
         }
+
+        [Fact]
+        public static void BadNamingPolicy_ThrowsInvalidOperation()
+        {
+            var options = new JsonSerializerOptions { DictionaryKeyPolicy = new NullNamingPolicy() };
```

Comment:
Can you add a test case with `PropertyNamingPolicy = new NullNamingPolicy()` as well, for completeness.

```C#
options = new JsonSerializerOptions { PropertyNamingPolicy = new NullNamingPolicy() };
Assert.Throws<InvalidOperationException>(() => JsonSerializer.Serialize(inputClass, options));
Assert.Throws<InvalidOperationException>(() => JsonSerializer.Deserialize("{}", options));
```

Also, add calls to Deserialize as well (not just serialize).

Reply:
For `DictionaryKeyPolicy` `Deserialize` is not using custom policy so not throwing, seems it was by design, https://github.com/dotnet/runtime/blob/de20ef43d2ef41ad00f20abdd247ef5ad99b0f5b/src/libraries/System.Text.Json/tests/Serialization/DictionaryTests.KeyPolicy.cs#L161

For PropertyNamingPolicy there is existng test https://github.com/dotnet/runtime/blob/de20ef43d2ef41ad00f20abdd247ef5ad99b0f5b/src/libraries/System.Text.Json/tests/Serialization/PropertyNameTests.cs#L63